lib/topology/platforms/shell.py

```diff
@@ -417,74 +428,70 @@ def get_response(self, connection=None):
 
         return '\n'.join(lines)
 
-    @_check_existing_connection
     def is_connected(self, connection=None):
         """
         See :meth:`BaseShell.is_connected` for more information.
         """
-        if connection is None:
-            return (
-                self._default_connection is not None
-            ) and self._connections[self._default_connection].isalive()
-        else:
-            return self._connections[connection].isalive()
+        # Get connection
+        spawn = self._get_connection(connection)
+        return spawn.isalive()
 
     def connect(self, connection=None):
         """
         See :meth:`BaseShell.connect` for more information.
         """
-        try:
-            if self.is_connected(connection=connection):
-                raise Exception('Shell already connected.')
-        except _NonExistingConnectionError:
-            pass
+        connection = connection or self._default_connection or '0'
 
-        # Evaluate connection value
-        if self._default_connection is None:
-            if connection is None:
-                connection = '0'
-            self._default_connection = connection
+        if connection in self._connections:
```

Comment:
Correct, we have a "reconnection" only in the sense that the same connection name is associated to a ``spawn`` object who is alive after the reconnection. I'll push a test for this and improved behavior documentation.